benchexec/tools/template.py

```diff
@@ -137,6 +137,7 @@ def version(self, executable):
     def _version_from_tool(
         executable,
         arg="--version",
+        stdin=None,
```

Comment:
@PhilippWendler When getting the version,
the tool somehow requires passing EOF (ctrl-D) to its stdin.
(Otherwise, it won't terminate.)

So I added a new argument to `BaseTool2._version_from_tool`
for specifying stdin.
Or would you prefer not touching the template?
In that case, I will move everything into BtorMC's `version`,
instead of calling `_version_from_tool`.

Reply:
Hm, this tool behavior is weird. If it requires EOF on stdin, then this should also mean that the tool similarly hangs when executed manually. Can you confirm this? If yes, I would suggest asking the tool developers to change this.

Independently, I wonder if we should change the behavior of BenchExec here. Without specifying stdin, the process inherits its stdin from the BenchExec process, i.e., it could end up receiving user input. This seems unwanted, confusing, and kind of breaks the isolation. Maybe we should just hardcode `subprocess.DEVNULL`? (Not in this PR, though.)

Reply:
> Hm, this tool behavior is weird. If it requires EOF on stdin, then this should also mean that the tool similarly hangs when executed manually.

The tool takes as input a Btor2 circuit via a file or stdin.
If a file is given, then it won't wait for stdin, i.e., won't hang.

Reply:
But it does seem like a mistake of the tool, because when specifying the other similar option `--copyright`,
BtorMC prints the message and terminates.
I'll ask the developer for clarification.

Reply:
I've implemented a fix myself and posed a [PR](https://github.com/Boolector/boolector/pull/225), but there is no response yet.

Reply:
I would like to evaluate this tool in our paper.
It would be nice if this PR could be merged such that I can use the main repository of BenchExec instead of my own fork,

Should I drop the `version` function from BtorMC's tool-info module for now?

Reply:
Given that it seems temporary, I wouldn't merge the current state of the PR with the changes to `template.py`. Because then I have to keep this API forever, and I am not sure I want that. But I could merge a PR with a `version` method if you do everything necessary there.

Reply:
> But I could merge a PR with a version method if you do everything necessary there.

Done

I also just noticed that my fix (https://github.com/Boolector/boolector/pull/225) has been approved.
I'll create another PR here after that has been merged.

Reply:
I now thought more about https://github.com/sosy-lab/benchexec/pull/1045#discussion_r1617030736 and decided that there seems to be no risk in using `/dev/null` as input for all tools when getting the version number. So I just did that in 9997760a495853458f8cbcc9b56ac8a8b3f9cf05. This should work for all versions of BtorMC if you just call the standard utility method. Would you mind trying that out and updating the PR?

Sorry for the back-and-forth, but this is the best overall solution I think.

Reply:
No problem at all!

I've tested 773f3d5 and it works.